### ISSUE TITLE
Defined parameters in Application getting overwritten

### ISSUE THREAD

**Issue by szucsitg** [argoproj-labs/argocd-image-updater#132]
**Describe the bug**
If you have a parameter in the Application, and there's a new image that's triggering an update of the Argo Application, it'll create an infinite sync loop.

**To Reproduce**
Add this to a Helm deployment's Application manifest:
```
spec:
  source:
    helm:
      parameters:
      - name: customVariable
        value: customValue
```

**Expected behavior**
Should not be an infinite loop of sync.

**Additional context**
N/A

**Version**
v0.8.0 - sha256:e31a6c1dce2cfbe12b84f1a68b6e6c85f4ca227c32a6c5bab20e2dc053457e2e

**Logs**
N/A

**Comment by jannfis** [comment]
Hi @szucsitg, I think this sort of sync loop can occur if you have an app-of-apps pattern, with autosync and self-heal turned on at the parent application. Is this true in your case?

**Comment by szucsitg** [comment]
Hello @jannfis,
We deploy the repository on the nth level of a dependency tree from another repo, and this repo is using Kustomize. All of our application are set with self-heal and prune.
If I turn off self-heal I guess this would fix the infinite loop, however in such case the Kustomize app which contains these Helm apps will be out of sync all the time.
Sidenote: also we use semver based auto-update for the Helm chart itself.

**Comment by jannfis** [comment]
Unfortunately, as long as [Git write-back](https://github.com/argoproj-labs/argocd-image-updater/issues/35) is not yet implemented, app-of-apps pattern is not supported, I fear. Probably even if you disable self-heal on the parent app, on the next sync of the parent app containing further `Application` resources, the changes made by Image Updater will be overwritten.

Good news is, that we already laid foundation for supporting parameters stored in application's Git repositor in Argo CD with https://github.com/argoproj/argo-cd/pull/4084 - now we have to connect the pieces, and make Image Updater push this file to Git. When this is done, also app-of-apps use cases will be supported.

Sorry that I can't offer you a quick solution in this case.

**Comment by szucsitg** [comment]
I didn't know about this YAML override, so i learned something new 👍  i guess it would not work either if we move the override parameter to `.argocd-app.yaml`. though I wonder how this YAML file would work in our case as we have currently multiple Application files within a folder.

**Comment by jannfis** [comment]
The YAML override is a "hidden" feature in Argo CD right now, which we implemented to later support Git write back from image updater in a non-intrusive way :)

There's a good point you've made. I think @alexmt and me discussed this scenario already (multiple apps spawned from same directory in Git), and came to the conclusion that we want to expand the YAML override to support this, i.e. by allowing application specific overrides through a file  `.argocd-app-<appname>.yaml` as well. But we wanted to start with something simple, just to see how this works out. But it's also good to see other use-cases, such as yours.

**Comment by szucsitg** [comment]
I'm not sure our pattern is ideal, but we have a couple of layers of dependency, Helm mixed with Kustomize back and forth, where we even loop through dynamically templated Argo apps (in Helm). Therefore many cases we have multiple apps within the same folder. Also most of our Kustomize app are multi-folder ones.

**Comment by jannfis** [comment]
I just created an enhancement proposal for Argo CD: https://github.com/argoproj/argo-cd/issues/5003

Feel free to contribute to this discussion :)

**Comment by jannfis** [comment]
With v0.9.0 being released (and hotfixed by v0.9.1), Git write-back is now available (with a strict dependency to Argo CD >= v2.0.0-rc1).
